src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -216,6 +216,134 @@ namespace AppInstaller::CLI::Workflow
+                // Check for an existing install and require a compatible locale.
+                auto installerLocaleItr = installationMetadata.find(Repository::PackageVersionMetadata::InstalledLocale);
+                if (installerLocaleItr != installationMetadata.end())
+                {
+                    return std::make_unique<InstalledLocaleComparator>(installerLocaleItr->second);
+                }
+
+                return {};
+            }
+
+            bool IsApplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                // We have to assume an unknown installer locale will match our installed locale, or the entire catalog would stop working for upgrade.
+                return installer.Locale.empty() || Utility::GetDistanceOfLanguage(m_installedLocale, installer.Locale) >= Utility::MinimumDistanceScoreAsCompatibleMatch;
+            }
+
+            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                std::string result = "Installer locale is not compatible with currently installed locale: ";
+                result += installer.Locale;
+                result += " not compatible with ";
+                result += m_installedLocale;
+                return result;
+            }
+
+            bool IsFirstBetter(const Manifest::ManifestInstaller& first, const Manifest::ManifestInstaller& second) override
+            {
+                double firstScore = first.Locale.empty() ? Utility::UnknownLanguageDistanceScore : Utility::GetDistanceOfLanguage(m_installedLocale, first.Locale);
+                double secondScore = second.Locale.empty() ? Utility::UnknownLanguageDistanceScore : Utility::GetDistanceOfLanguage(m_installedLocale, second.Locale);
+
+                return firstScore > secondScore;
+            }
+
+        private:
+            std::string m_installedLocale;
+        };
+
+        struct LocaleComparator : public details::ComparisonField
+        {
+            LocaleComparator(std::string preference, std::string requirement) :
+                details::ComparisonField("Locale"), m_preference(std::move(preference)), m_requirement(std::move(requirement)) {}
+
+            static std::unique_ptr<LocaleComparator> Create(const Execution::Args& args)
+            {
+                std::string preference;
+                std::string requirement;
+
+                // Preference will come from winget settings or Preferred Languages settings. winget settings takes precedence.
+                preference = Settings::User().Get<Settings::Setting::InstallLocalePreference>();
+                if (preference.empty())
+                {
+                    auto preferredList = Utility::GetUserPreferredLanguages();
+                    if (!preferredList.empty())
+                    {
+                        // TODO: we only take the first one for now
+                        preference = preferredList.at(0);

```

Comment:
This suggests that the setting should also be an array.  What are the complications with taking the whole list?  The only one that comes to mind is that the sort is harder, but I don't think that is insurmountable.